packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -781,3 +800,108 @@ void _exitWithExpectedFileNotFound({
     'but the tool couldn\'t find it.'
   );
 }
+
+void _createSymlink(String targetPath, String linkPath) {
+  final File targetFile = fs.file(targetPath);
+  if (!targetFile.existsSync()) {
+    throwToolExit('The file $targetPath wasn\'t found in the local engine out directory.');
+  }
+  final File linkFile = fs.file(linkPath);
+  final Link symlink = linkFile.parent.childLink(linkFile.basename);
+  try {
+    symlink.createSync(targetPath, recursive: true);
+  } on FileSystemException catch (exception) {
+    throwToolExit(
+      'Failed to create the symlink $linkPath->$targetPath: $exception'
+    );
+  }
+}
+
+String _getLocalArtifactVersion(String pomPath) {
+  final File pomFile = fs.file(pomPath);
+  if (!pomFile.existsSync()) {
+    throwToolExit('The file $pomPath wasn\'t found in the local engine out directory.');
+  }
+  xml.XmlDocument document;
+  try {
+    document = xml.parse(pomFile.readAsStringSync());
+  } on xml.XmlParserException {
+    throwToolExit(
+      'Error parsing $pomPath. Please ensure that this is a valid XML document.'
+    );
+  } on FileSystemException {
+    throwToolExit(
+      'Error reading $pomPath. Please ensure that you have read permission to this'
+      'file and try again.');
+  }
+  final Iterable<xml.XmlElement> project = document.findElements('project');
+  assert(project.isNotEmpty);
+  for (xml.XmlElement versionElement in document.findAllElements('version')) {
+    if (versionElement.parent == project.first) {
+      return versionElement.text;
+    }
+  }
+  throwToolExit('Error while parsing the <version> element from $pomPath');
+  return null;
+}
+
+/// Returns the local Maven repository for a local engine build.
+/// For example, if the engine is built locally at <home>/engine/src/out/android_release_unopt
+/// This method generates symlinks in the temp directory to the engine artifacts
+/// following the convention specified on https://maven.apache.org/pom.html#Repositories
+Directory _getLocalEngineRepo({
+  @required String engineOutPath,
+  @required AndroidBuildInfo androidBuildInfo,
+}) {
+  assert(engineOutPath != null);
+  assert(androidBuildInfo != null);
+
+  final String abi = getEnumName(androidBuildInfo.targetArchs.first);
+  final Directory localEngineRepo = fs.systemTempDirectory
+    .createTempSync('local_engine_repo.');
```

Comment:
nit: temp directories should be named something like flutter_tool_X, see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#have-good-hygiene-when-using-temporary-directories

Reply:
Done